### ISSUE TITLE
P2D renderer fails displaying objects past 32k units translated.

### ISSUE THREAD

**Issue by protonlaser91** [processing/processing#6174]
## Description
P2D renderer unable to draw objects past 32k translation

## Expected Behavior
 The objects that are being drawn should continue to be drawn regardless of the translation.

## Current Behavior

At around 32,768 units translated, the sketch no longer displays the object and instead displays a large box (or in some instances the object just begins to cut out as if it is being cropped).

## Steps to Reproduce
```processing
void setup() {
  size(800, 800, P2D); // The third argument can be replaced with 
  smooth(8);
   
}

PShape l;

float f(float x) {
  return 100*sin((x-3*frameCount)/100f);
}

void draw() {
  
  background(0);
  translate(width/2 + (-200*frameCount),height/2); // Translating (the objects being drawn are offset by the same amount so they remain in scene).
  
  l = createShape();

  l.beginShape(LINES);
   l.stroke(0,255,0);
   l.strokeWeight(6);
   for (int i = -100; i < 100; i++) { // Removing this seems to just crop out the remaining line as the sketch approaches 32k.
    l.vertex(i + 200*frameCount, f(i));
  }
  l.vertex(200*frameCount - 300,0);
  l.vertex(200*frameCount + 300, 0);

   l.endShape();
   shape(l);

 
  println(200*frameCount);
}
```

## Your Environment

 

* Processing version: 3.5.4
* Operating System and OS version: Windows 10 Version 2004 (Build 19041.867).
* Other information:

## Possible Causes / Solutions

I did further testing with this and it seems that zooming out will show the objects as normal. 

![ezgif com-gif-maker (10)](https://user-images.githubusercontent.com/44213272/114319100-060f1100-9ade-11eb-91d3-7a47518bbb0a.gif)
![ezgif com-gif-maker (9)](https://user-images.githubusercontent.com/44213272/114319101-06a7a780-9ade-11eb-93f8-9df4fc382a68.gif)

**Comment by benfry** [comment]
I've answered this a couple times, but just added a note to the Troubleshooting FAQ page:

    If you're using coordinates larger than (or near) 16,384 (2^15) or 32,768 (2^16), 
    you're likely to have problems. This is because of low-level hardware 
    and operating system limits, and not something that we can control.
    In fact, it's *so* out of our control that it's difficult when you'll
    hit these issues: what operating systems, what hardware, etc. 
    Bottom line: just use smaller numbers.
